test/Sema/diag_subtype_argument_representation.swift

```diff
@@ -0,0 +1,95 @@
+// RUN: %target-typecheck-verify-swift -emit-sil
```

Comment:
@xedin Although this is now on Sema, I wanted a way to ensure those cases will not crash at SILGen... is this ok?
Another option I thought was to put the same cases with `-emit-sil` under compiler_crashes_fixed in validation